### PR TITLE
program name as window title, Qt and OpenGL fixes

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

**Comment by tribal-tec** [comment]
Missing changelog update

**Comment by eile** [comment]
Changelog: https://github.com/Eyescale/Equalizer/pull/463/files#diff-0fbe52982afdae190d190551546b7e24

**Comment by tribal-tec** [comment]
After the release

File: CMake/FindPackages.cmake [hunk 1]
```diff
@@ -58,7 +58,9 @@ common_package(Collage 1.1  REQUIRED )
 common_package(OpenGL   REQUIRED )
 common_package(Boost 1.41.0  REQUIRED COMPONENTS program_options filesystem system thread)
 common_package(Pression 1.0  REQUIRED )
-common_package(X11    )
+if(NOT EQ_AGL_USED)
```
Comment (tribal-tec) [inline CMake/FindPackages.cmake]: will be overwritten by Buildyard

File: CMake/FindPackages.cmake [hunk 2]
```diff
@@ -68,9 +70,11 @@ common_package(VRPN 07.30   )
 common_package(Deflect 0.4   )
 common_package(MAGELLAN    )
 common_package(GLEW_MX    )
-common_package(Qt5Core    )
-common_package(Qt5Gui    )
-common_package(Qt5Widgets    )
+if(NOT X11_FOUND)
```
Comment (tribal-tec) [inline CMake/FindPackages.cmake]: dito.
besides, this means no Qt on Linux??

File: doc/RelNotes.md [hunk 3]
```diff
@@ -39,6 +39,7 @@ Equalizer provides the following major features to facilitate the development an
 
 ## git master
 
+* Use program name as default window title
```
Comment (tribal-tec) [inline doc/RelNotes.md]: put to new Changelog.md

File: eq/client.h [hunk 4]
```diff
@@ -105,6 +105,17 @@ class Client : public fabric::Client
     /** @internal @return the command queue to the main node thread. */
     EQ_API virtual co::CommandQueue* getMainThreadQueue();
 
+    /** Experimental: interrupt main thread queue @internal */
+    void interruptMainThread();
+
+    /** @name Data Access */
+    //@{
+    /** Set the name of the configuration. @version 1.8 */
```
Comment (tribal-tec) [inline eq/client.h]: 1.10

File: eq/client.h [hunk 5]
```diff
@@ -105,6 +105,17 @@ class Client : public fabric::Client
...
+    EQ_API void setName( const std::string& name );
+
+    /** @return the name of the configuration. @version 1.8 */
```
Comment (tribal-tec) [inline eq/client.h]: 1.10

File: eq/fabric/configParams.h [hunk 6]
```diff
@@ -73,6 +73,12 @@ class ConfigParams
 
     /** @name Data Access. */
     //@{
+    /** Set the name of the configuration. @version 1.8 */
```
Comment (tribal-tec) [inline eq/fabric/configParams.h]: 1.10

File: eq/fabric/configParams.h [hunk 7]
```diff
@@ -73,6 +73,12 @@ class ConfigParams
...
+    EQFABRIC_API void setName( const std::string& name );
+
+    /** @return the name of the configuration. @version 1.8 */
```
Comment (tribal-tec) [inline eq/fabric/configParams.h]: 1.10

**Comment by tribal-tec** [comment]
rebase

File: CMake/FindPackages.cmake (hunk 1 above)
Comment (eile) [inline CMake/FindPackages.cmake]: I know. It's a placeholder for me, will do properly once your cmake changes are in.

File: CMake/FindPackages.cmake (hunk 2 above)
Comment (eile) [inline CMake/FindPackages.cmake]: good point. will fix.

Comment (tribal-tec) [inline CMake/FindPackages.cmake]: where is the fix?

File: doc/Changelog.md [hunk 8]
```diff
@@ -1,6 +1,12 @@
 # Changelog
 
-# Release 1.9 (07-07-2015) {#Release19}
+## git master {#master}
+
+* Use program name as default window title
```
Comment (tribal-tec) [inline doc/Changelog.md]: link to PR

File: doc/Changelog.md [hunk 9]
```diff
@@ -1,6 +1,12 @@
...
+* Implemented eq::x11::Window and eqCPU example as a template for
+  CPU-based rendering, for example for raytracing
```
Comment (tribal-tec) [inline doc/Changelog.md]: was already part of previous release: https://github.com/Eyescale/Equalizer/pull/437

**Comment by tribal-tec** [comment]
Version bump to 1.10

File: CMake/FindPackages.cmake (hunk 2 above)
Comment (eile) [inline CMake/FindPackages.cmake]: Not yet submitted. Still working on the Qt on OS X stuff.

**Comment by eile** [comment]
retest this please

**Comment by eile** [comment]
Can we start reviewing this before it becomes even bigger?

I'll work on Jenkins in parallel.

**Comment by eile** [comment]
retest this please.

File: CMake/FindPackagesPost.cmake [hunk 10]
```diff
@@ -34,32 +30,43 @@ if(APPLE)
...
-    set(EQ_EXAMPLES_OSX_ARCHITECTURES ${CMAKE_OSX_ARCHITECTURES})
+  set(EQ_EXAMPLES_OSX_ARCHITECTURES ${CMAKE_OSX_ARCHITECTURES})
+endif()
+
+if(X11_FOUND)
+  if(EQ_AGL_USED)
+    set(X11_FOUND)
+  else()
+    set(EQ_GLX_USED 1)
+    list(APPEND FIND_PACKAGES_DEFINES EQUALIZER_USE_X11)
```
Comment (tribal-tec) [inline CMake/FindPackagesPost.cmake]: this variable name changed

File: CMake/FindPackagesPost.cmake [hunk 11]
```diff
@@ -34,32 +30,43 @@ if(APPLE)
...
+    set(FIND_PACKAGES_FOUND "${FIND_PACKAGES_FOUND} X11")
```
Comment (tribal-tec) [inline CMake/FindPackagesPost.cmake]: dito iirc

File: CMakeLists.txt [hunk 12]
```diff
@@ -24,6 +24,7 @@ option(EQUALIZER_USE_HWLOC "Test for hwloc support (experimental)" OFF)
 option(EQUALIZER_INSTALL_SERVER_HEADERS "Install Equalizer server headers" OFF)
 option(EQUALIZER_BUILD_2_0_API
   "Enable for pure 2.0 API (breaks compatibility with 1.x API)" OFF)
+option(EQUALIZER_PREFER_QT "Prefer Qt window system over X11 on OS X" OFF)
```
Comment (tribal-tec) [inline CMakeLists.txt]: why not ON?

File: CMakeLists.txt [hunk 13]
```diff
@@ -38,13 +39,16 @@ if(EQUALIZER_BUILD_2_0_API)
...
+  set(EQ_AGL_USED 1)
+endif()
 
 include(configure)
 
 list(APPEND CPPCHECK_EXTRA_ARGS -DEQUALIZERFABRIC_STATIC=)
 
 if(CMAKE_COMPILER_IS_CLANG)
-  set(COMMON_COMPILER_CXX_FLAGS "${COMMON_COMPILER_CXX_FLAGS} -Wno-overloaded-virtual")
+  set(COMMON_CXX_FLAGS "${COMMON_CXX_FLAGS} -Wno-overloaded-virtual")
```
Comment (tribal-tec) [inline CMakeLists.txt]: comment again why we need this flag?

File: GLEW/CMakeLists.txt [hunk 14]
```diff
@@ -38,9 +38,13 @@ else()
 
   include_directories(BEFORE SYSTEM ${GLEW_MX_INCLUDE_DIRS})
   add_definitions(-DGLEW_MX=1 -DGLEW_NO_GLU)
-  add_library(${GLEW_LIBRARY} ${GLEW_DIR}/src/glew.c)
+  add_library(${GLEW_LIBRARY} SHARED ${GLEW_DIR}/src/glew.c)
```
Comment (tribal-tec) [inline GLEW/CMakeLists.txt]: common_library would work?

File: eq/agl/pipe.h [hunk 15]
```diff
@@ -29,64 +29,59 @@ namespace eq
...
-
-        /**
-         * De-initialize this pipe for the AGL window system.
-         *
-         * @return true if the deinitialization was successful, false otherwise.
-         * @version 1.0
-         */
-        virtual void configExit( );
-        //@}
+    virtual bool configInit( );
```
Comment (tribal-tec) [inline eq/agl/pipe.h]: final

File: eq/agl/pipe.h [hunk 16]
```diff
@@ -29,64 +29,59 @@ namespace eq
...
-        CGDirectDisplayID getCGDisplayID() const { return _cgDisplayID; }
-
-    private:
+    /**
+     * De-initialize this pipe for the AGL window system.
+     *
+     * @return true if the deinitialization was successful, false otherwise.
+     * @version 1.0
+     */
+    virtual void configExit( );
```
Comment (tribal-tec) [inline eq/agl/pipe.h]: dito

File: CMakeLists.txt (hunk 12 above)
Comment (eile) [inline CMakeLists.txt]: It's not as complete as X11 (no fonts). I don't have a strong opinion, can switch it to ON if you prefer.

File: eq/client.cpp [hunk 17]
```diff
@@ -64,14 +70,33 @@ class Client
...
     Strings activeLayouts;
     ServerSet localServers;
     std::string gpuFilter;
     float modelUnit;
     bool running;
+    QApplication* qtApp;
+
+    void initQt( int argc LB_UNUSED, char** argv LB_UNUSED )
+    {
+#if EQ_GLX_USED || EQ_WGL_USED || EQ_AGL_USED
```
Comment (tribal-tec) [inline eq/client.cpp]: why? If not EQ_QT_USED, it has the same effect.

File: eq/qt/windowSystem.cpp [hunk 18]
```diff
@@ -79,25 +64,15 @@ std::string WindowSystem::getName() const
 eq::SystemWindow* WindowSystem::createWindow( eq::Window* window,
                                               const WindowSettings& settings )
 {
-    if( _useSystemWindowSystem( settings, window->getSharedContextWindow()))
-        return getSystemWindowSystem().createWindow( window, settings );
-
-    QOpenGLContext* context = _getShareContext( settings );
-
     LBINFO << "Using qt::Window" << std::endl;
```
Comment (tribal-tec) [inline eq/qt/windowSystem.cpp]: LBDEBUG

**Comment by tribal-tec** [comment]
Yes, that's a big change. Not more please :) But looks very good. I'd always wanted to work on the Qt for OSX, but I'm curious to use it too!

File: CMakeLists.txt (hunk 12 above)
Comment (tribal-tec) [inline CMakeLists.txt]: Ah, true, the fonts... Leave it then.

File: eq/client.cpp (hunk 17 above)
Comment (eile) [inline eq/client.cpp]: No, I only want to create an app if I'm in Qt-only (not Qt+AGL/GLX) mode.

**Comment by eile** [comment]
Not my fault with the big change. Half of it was ready last week (and waiting on the CMake stuff). Will address the rest.

**Comment by tribal-tec** [comment]
Yes, your fault. The CMake stuff was planned, OSX Livre was not :p

**Comment by eile** [comment]
retest this please

**Comment by tribal-tec** [comment]
+1 now. Merge when green.

**Comment by eile** [comment]
retest this please

**Pull request merged**